src/stream/src/executor/source/source_backfill_executor.rs

```diff
@@ -727,6 +766,18 @@ impl<S: StateStore> SourceBackfillExecutorInner<S> {
             debug_assert_eq!(old_states, target_state);
         }
         stage.states = target_state;
+        stage.splits = target_splits;
```

Comment:
This line fixed a bug https://github.com/risingwavelabs/risingwave/pull/18296/files#r1738292017